### PR TITLE
Feat strict types

### PR DESCRIPTION
**Pull request by alexander-akait**
**What kind of change does this PR introduce?**

types

**Did you add tests for your changes?**

No need

**If relevant, did you update the documentation?**

No need

**Summary**

Use `strict`

**Does this PR introduce a breaking change?**

No

**Other information**

- refactor our code
- remove unused things
- avoid extra any

there are still a lot of any, so still WIP, also we need to refactor our generators more, to use one CustomGenerator class

### CODE REVIEW

**Comment by alexander-akait** [comment]
@webpack/cli-team Can you review, I rewrite our logic on using only one custom generator, so we don't need copy/paste logic, also improve types, now we use `strict`, any `cli` itself is `any`, but we will finish it in https://github.com/webpack/webpack-cli/pull/3076

File: packages/generators/src/handlers/default.ts [hunk 1]
```diff
@@ -198,7 +192,6 @@ export function generate(self: CustomGenerator): void {
     ...self.answers,
     entry,
   });
-  self.configurationPath = self.destinationPath("webpack.config.js");
```
Comment (rishabh3112) [inline packages/generators/src/handlers/default.ts:201]: Not a big deal for now, but IMO, This should be handled here only, as templates might generate multiple configuration files as well in future.

**Review by rishabh3112 — APPROVED** [review]
Looks good, Just one doubt.

File: packages/generators/src/handlers/default.ts (hunk 1 above)
Comment (alexander-akait) [reply packages/generators/src/handlers/default.ts:201]: Yep, let's do refactor when we will implement it, I think now it good, focus on types for webpack-cli

**Pull request merged**